bemani/frontend/mga/endpoints.py

```diff
@@ -0,0 +1,175 @@
+def listplayers() -> Dict[str, Any]:
+    frontend = MetalGearArcadeFrontend(g.data, g.config, g.cache)
+    return {
+        'players': frontend.get_all_players(),
+    }
+
+
+@mga_pages.route('/players/<int:userid>')
+@loginrequired
+def viewplayer(userid: UserID) -> Response:
+    frontend = MetalGearArcadeFrontend(g.data, g.config, g.cache)
+    djinfo = frontend.get_all_player_info([userid])[userid]
+    if not djinfo:
+        abort(404)
+    latest_version = sorted(djinfo.keys(), reverse=True)[0]
+
+    return render_react(
+        f'{djinfo[latest_version]["name"]}\'s MGA Profile',
+        'mga/player.react.js',
+        {
+            'playerid': userid,
+            'own_profile': userid == g.userID,
+            'player': djinfo,
+            'versions': {version: name for (game, version, name) in frontend.all_games()},
+        },
+        {
+            'refresh': url_for('mga_pages.listplayer', userid=userid),
+        },
+    )
+
+
+@mga_pages.route('/players/<int:userid>/list')
+@jsonify
+@loginrequired
+def listplayer(userid: UserID) -> Dict[str, Any]:
+    frontend = MetalGearArcadeFrontend(g.data, g.config, g.cache)
+    djinfo = frontend.get_all_player_info([userid])[userid]
+
+    return {
+        'player': djinfo,
+    }
+
+
+@mga_pages.route('/options')
+@loginrequired
+def viewsettings() -> Response:
+    frontend = MetalGearArcadeFrontend(g.data, g.config, g.cache)
+    userid = g.userID
+    djinfo = frontend.get_all_player_info([userid])[userid]
+    if not djinfo:
+        abort(404)
+
+    return render_react(
+        'Metal Gear Arcade Game Settings',
+        'mga/settings.react.js',
+        {
+            'player': djinfo,
+            'versions': {version: name for (game, version, name) in frontend.all_games()},
+        },
+        {
+            'updatename': url_for('mga_pages.updatename'),
+        },
+    )
+
+
+@mga_pages.route('/options/name/update', methods=['POST'])
+@jsonify
+@loginrequired
+def updatename() -> Dict[str, Any]:
+    frontend = MetalGearArcadeFrontend(g.data, g.config, g.cache)
+    version = int(request.get_json()['version'])
+    name = request.get_json()['name']
+    user = g.data.local.user.get_user(g.userID)
+    if user is None:
+        raise Exception('Unable to find user to update!')
+
+    # Grab profile and update dj name
+    profile = g.data.local.user.get_profile(GameConstants.MGA, version, user.id)
+    if profile is None:
+        raise Exception('Unable to find profile to update!')
+    if len(name) == 0 or len(name) > 6:
+        raise Exception('Invalid profile name!')
+
+    # Convert lowercase to uppercase. We allow lowercase widetext in
+    # the JS frontend to allow for Windows IME input of hiragana/katakana.
+    def conv(char: str) -> str:
+        i = ord(char)
+        if i >= 0xFF41 and i <= 0xFF5A:
+            return chr(i - (0xFF41 - 0xFF21))
+        else:
+            return char
+    name = ''.join([conv(a) for a in name])
+
+    if re.match(
```

Comment:
Is this verified or just copied from Bishi? If the latter, could you put a TODO here to verify the regex?